Tests/JExtractSwiftTests/UnsignedNumberTests.swift

```diff
@@ -0,0 +1,268 @@
+  func return_unsignedLong_annotate() throws {
+    var config = Configuration()
+    config.unsignedNumbersMode = .annotate
+
+    try assertOutput(
+      input: "public func returnUnsignedLong() -> UInt64",
+      config: config,
+      .ffm, .java,
+      detectChunkByInitialLines: 2,
+      expectedChunks: [
+        """
+        /**
+         * {@snippet lang=c :
+         * uint64_t swiftjava_SwiftModule_returnUnsignedLong(void)
+         * }
+         */
+        private static class swiftjava_SwiftModule_returnUnsignedLong {
+          private static final FunctionDescriptor DESC = FunctionDescriptor.of(
+          /* -> */SwiftValueLayout.SWIFT_UINT64
+        );
+        """,
+        """
+        @Unsigned
+        public static long returnUnsignedLong() {
+          return swiftjava_SwiftModule_returnUnsignedLong.call();
+        }
+        """,
+      ]
+    )
+  }
+
+  @Test("Import: take UInt64 (annotate)")
+  func take_unsignedLong_annotate() throws {
+    var config = Configuration()
+    config.unsignedNumbersMode = .annotate
+
+    try assertOutput(
+      input: "public func takeUnsignedLong(arg: UInt64)",
+      config: config,
+      .ffm, .java,
+      detectChunkByInitialLines: 2,
+      expectedChunks: [
+        """
+        /**
+         * {@snippet lang=c :
+         * void swiftjava_SwiftModule_takeUnsignedLong_arg(uint64_t arg)
+         * }
+         */
+        private static class swiftjava_SwiftModule_takeUnsignedLong_arg {
+          private static final FunctionDescriptor DESC = FunctionDescriptor.ofVoid(
+          /* arg: */SwiftValueLayout.SWIFT_UINT64
+        );
+        """,
+        """
+        public static void takeUnsignedLong(@Unsigned long arg) {
+          swiftjava_SwiftModule_takeUnsignedLong_arg.call(arg);
+        }
+        """,
+      ]
+    )
+  }
+
+  @Test("Import: take UInt64 return UInt32 (annotate)")
+  func echo_unsignedLong_annotate() throws {
+    var config = Configuration()
+    config.unsignedNumbersMode = .annotate
+
+    try assertOutput(
+      input: "public func unsignedLong(first: UInt64, second: UInt32) -> UInt32",
+      config: config,
+      .ffm, .java,
+      detectChunkByInitialLines: 2,
+      expectedChunks: [
+        """
+        /**
+         * {@snippet lang=c :
+         * uint32_t swiftjava_SwiftModule_unsignedLong_first_second(uint64_t first, uint32_t second)
+         * }
+         */
+        private static class swiftjava_SwiftModule_unsignedLong_first_second {
+          private static final FunctionDescriptor DESC = FunctionDescriptor.of(
+          /* -> */SwiftValueLayout.SWIFT_UINT32
+          /* first: */SwiftValueLayout.SWIFT_UINT64
+          /* second: */SwiftValueLayout.SWIFT_UINT32
+        );
+        """,
+        """
+        @Unsigned
+        public static int unsignedLong(@Unsigned long first, @Unsigned int second) {
+          return swiftjava_SwiftModule_unsignedLong_first_second.call(first, second);
+        }
```

Comment:
Showcase of the default extract mode -- we don't do any conversions, but we annotate the types for easier discovery in IDEs

<img width="427" height="78" alt="Image" src="https://github.com/user-attachments/assets/6b65dd7a-a19a-4ab0-ba23-cc8b65ab596d" />